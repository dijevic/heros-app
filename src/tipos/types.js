export const types = {
    login: '[auth] login',
    logOut: `[auth] logOut`
}