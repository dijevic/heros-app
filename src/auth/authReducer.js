import { types } from "../tipos/types";

export const authReducer = (state = {}, action) => {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logOut:
            return { logged: false }

        default:
            return state
    }
}