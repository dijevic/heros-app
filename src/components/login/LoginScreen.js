import React, { useContext } from 'react'
import { authContext } from '../../auth/AuthContext'
import { types } from '../../tipos/types'

export const LoginScreen = ({ history }) => {
    const { dispatch } = useContext(authContext)
    const handleClick = () => {
        const path = localStorage.getItem('lastPath') || '/'
        const user = {
            nombre: 'diego jesus',
            apellido: 'vielma carrero',
            id: 123456,
        }
        dispatch({
            type: types.login,
            payload: user
        })
        history.replace(path)



    }
    return (
        <div className="container mt-5">
            <h1>Login screen</h1>

            <button className="btn btn-primary" onClick={handleClick}>Ingresar</button>
        </div>
    )
}
