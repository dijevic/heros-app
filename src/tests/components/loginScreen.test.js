import { mount } from "enzyme"
import { authContext } from "../../auth/AuthContext"
import { LoginScreen } from "../../components/login/LoginScreen"
import { types } from "../../tipos/types"

describe('pruebas en mi componente del login', () => {

    Storage.prototype.getItem = jest.fn()
    const history = {
        replace: jest.fn()
    }
    const context = {
        dispatch: jest.fn()
    }
    const wrapper = mount(
        <authContext.Provider value={context}>
            <LoginScreen history={history} />
        </authContext.Provider>)
    test('debe renderizar correctamente', () => {


        expect(wrapper).toMatchSnapshot()
    })

    test('debo llamar a mi dispatch al hacer click', () => {
        const user = {
            nombre: 'diego jesus',
            apellido: 'vielma carrero',
            id: 123456,
        }
        wrapper.find('button').simulate('click')
        expect(context.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: user
        })

        expect(history.replace).toHaveBeenCalledWith('/')

        expect(localStorage.getItem).toHaveBeenCalled()


    })
})