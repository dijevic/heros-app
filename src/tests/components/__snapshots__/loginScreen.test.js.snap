// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en mi componente del login debe renderizar correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login screen
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Ingresar
  </button>
</div>
`;
